pkg/controller/operators/olm/operator.go

```diff
@@ -667,3 +730,56 @@ func (a *Operator) isReplacing(in *v1alpha1.ClusterServiceVersion) *v1alpha1.Clu
 	}
 	return previous
 }
+
+func (a *Operator) handleDeletion(obj interface{}) {
+	ownee, ok := obj.(metav1.Object)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			return
+		}
+
+		ownee, ok = tombstone.Obj.(metav1.Object)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a Namespace %#v", obj))
+			return
+		}
+	}
+
+	a.requeueOwnerCSVs(ownee)
+}
+
+func (a *Operator) requeueOwnerCSVs(ownee metav1.Object) {
+	logger := log.WithFields(log.Fields{
+		"ownee":    ownee.GetName(),
+		"selflink": ownee.GetSelfLink(),
+	})
+
+	// Get all existing CSVs from the indexer
+	csvs, err := a.lister.OperatorsV1alpha1Lister().ClusterServiceVersionLister().List(labels.Everything())
```

Comment:
I think going by UID and looking through all CSVs on the cluster is the best way to handle requeueing in the general case (Since cluster level resources can have ownerreferences to CSVs in multiple namespaces). Using the indexer means we don't need to query the Kube API, and if by chance a CSV doesn't exist in the indexer cache yet, it will be by at most the time of the next resync.

Reply:
I think I'm missing why we would bother listing and checking for existence here? requeueing is fast (adds a key to the queue), and we'll detect that it's not there anymore when we pop it. The first thing queue processing does is check if the object exists in the cache.

Reply:
We are listing so we can requeue owner CSVs by the UID listed in the object's OwnerReferences. We need to do it this way because objects like ClusterRoles and ClusterRoleBindings have no context to figure out which namespace their owner CSV hails from. Requeueing is quick, but why not avoid requeueing all CSVs on the cluster every time? The "existence" check is really just a byproduct of creating a set to efficiently find which CSVs to requeue anyway.

Reply:
In that case, I think this should only fall back to using the lister, e.g.:

```
 if owner ref has name/namespace:
    requeue(name, namespace)
 if owner ref missing namespace:
    list csvs
    find namespace
    requeue(name, namespace)
```